src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/DynamicExpressionUtilities.cs

```diff
@@ -0,0 +1,289 @@
+﻿using Microsoft.DotNet.Interactive.Http;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Interactive.Http.Parsing
+{
+    #nullable enable   
+    internal class DynamicExpressionUtilites
+    {
+        internal static HttpBindingResult<object?> MatchExpressionValue(HttpExpressionNode node, string expression)
+        {
+            const string DateTime = "$datetime";
+            const string LocalDateTime = "$localDatetime";
+            const string OffsetRegex = """(?:\s+(?<offset>[-+]?[^\s]+)\s+(?<option>[^\s]+))?""";
+            const string TypeRegex = """(?:\s+(?<type>rfc1123|iso8601|'.+'|".+"))?""";
+
+            var guidPattern = new Regex(@$"^\$guid$", RegexOptions.Compiled);
+            var dateTimePattern = new Regex(@$"^\{DateTime}{TypeRegex}{OffsetRegex}$", RegexOptions.Compiled);
+            var localDateTimePattern = new Regex(@$"^\{LocalDateTime}{TypeRegex}{OffsetRegex}$", RegexOptions.Compiled);
+            var randomIntPattern = new Regex(@$"^\$randomInt(?:\s+(?<arguments>-?[^\s]+)){{0,2}}$", RegexOptions.Compiled);
+            var timestampPattern = new Regex(@$"^\$timestamp{OffsetRegex}$", RegexOptions.Compiled);
```

Comment:
These can be initialized as static fields instead of local variables, since they're effectively immutable. That's also where `RegexOptions.Compiled` will give an advantage.